docs/docs/cmd/teams/message/message-export.mdx

```diff
@@ -0,0 +1,170 @@
+import Global from '/docs/cmd/_global.mdx';
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# teams message export
+
+Exports Microsoft Teams chat messages for a given user or a team.
+
+## Usage
+
+```sh
+m365 teams message export [options]
+```
+
+## Options
+
+```md definition-list
+`-u, --userId [userId]`
+: The ID of the user whose messages are to be exported. Specify either `userId`, `userName` or `teamId`
+
+`-n, --userName [userName]`
+: The UPN of the user whose messages are to be exported. Specify either `userId`, `userName` or `teamId`.
+
+`-i, --teamId [teamId]`
+: The ID of the team of which the messages are to be exported. Specify either `userId`, `userName` or `teamId`.
+
+`-f, --fromDateTime [fromDateTime]`
+: The start of the time range to query. This should be defined as a valid ISO 8601 string. `2021-12-16T18:28:48.6964197Z`
+
+`-t --toDateTime [toDateTime]`
+: The end of the time range to query. This should be defined as a valid ISO 8601 string. `2021-12-16T18:28:48.6964197Z`
+
+`-a, --withAttachments`
+: When specified, the command will also download all chat attachments. Use `folderPath` as well to set the rootfolder.
+
+`-p, --folderPath [folderPath]`
+: The folder path to save downloaded attachments to when using `withAttachments`. A specific folder will be created to store the attachments for every message.
+```
+
+<Global />
+
+## Remarks
+
+Microsoft Teams APIs in Microsoft Graph that access sensitive data are considered protected APIs. Export APIs require that you have additional validation, beyond permissions and consent, before you can use them. To access the protected APIs, please refer to the following url: [Enable metered APIs and services in Microsoft Graph](https://learn.microsoft.com/en-us/graph/metered-api-setup?tabs=azurecloudshell)
```

Comment:
Hi @MathijsVerbeeck, Being content about the code quality (😁), I'm now trying this out. I'm getting an error saying `Error: Requested API is not supported in delegated context`. Should we add a message about this in the docs?

Reply:
Great remark. I'll add a remark regarding this, and probably best add a check when we call the command and throw a user friendly error?

Reply:
Yes, let's do that as well!

Reply:
By the way: when I used withAttachment I got a 401, but that was because I was using a client Id and secret, which the SharePoint api does not support (need to use certificate there)

I see why you used the SharePoint api there: you only had a url and a file id. I think using the graph api for attachments would be more consistent within the command. But I don't think that will work without additional requests, is that also your finding?

Reply:
It's strange, because that seemed to work on my end.

I'll have to do a check on how I authenticated to use the command, because I was able to easily export all my attachments, located anywhere, and did it using the SharePoint API's. I'll get back to you regarding this.

Reply:
@martinlingstuyl I just did a check and I authenticate using `--authType certificate` as you say. 

I'll change the functionality so that it will work with Graph in the future.

Reply:
Are you saying that it's possible to use the graph api here?

Reply:
I think that it should be, yes. but I'll have to do some research how these endpoints work exactly, as we will also be dealing with personal sites.